src/lib/releaseTestPlanGenerator.ts

```diff
@@ -0,0 +1,171 @@
+               mergedTestCases
+            )
+         );
+      });
+
+   const mergedTestPlan: TestPlan = {
+      testCases: mergedTestCases,
+      name: filename,
+      versionToTest: versionToTest,
+   };
+   return await exportTestPlan(mergedTestPlan);
+}
+
+/**
+ * Merge a Test Case when it has already been required by another Test Plan
+ */
+export function mergeTestCase(
+   testCase: TestCase,
+   requiredByTestPlanName: string,
+   mergedTestCases: TestCase[]
+): void {
+   // Test case does not exists, add it
+   if (!testCaseExists(testCase, mergedTestCases)) {
+      var newRequiredBy = testCase.isRequired ? requiredByTestPlanName : "";
+
+      mergedTestCases.push({
+         featureName: testCase.featureName,
+         isRequired: testCase.isRequired,
+         requiredBy: newRequiredBy,
+      });
+   } else {
+      // Test case exists, requires merging
+      const existingTestCaseIndex = findIndexForTestCase(
+         testCase,
+         mergedTestCases
+      );
+
+      if (testCase.isRequired) {
+         if (mergedTestCases[existingTestCaseIndex].isRequired) {
+            // Existing test case was required by another test plan, augment requiredBy with this one
+            mergedTestCases[existingTestCaseIndex].requiredBy =
+               mergedTestCases[existingTestCaseIndex].requiredBy +
+               " " +
+               requiredByTestPlanName;
+         } else {
+            // Existing test case was not previously required, add only this requireBy
+            mergedTestCases[
+               existingTestCaseIndex
+            ].requiredBy = requiredByTestPlanName;
+         }
+
+         // Ensure the test case is now required
+         mergedTestCases[existingTestCaseIndex].isRequired = true;
+      }
+   }
+}
+
+/**
+ * Load the content of a Test Plan that was previously saved to cvs file
+ * @param filename
+ */
+export async function loadTestPlanFromFile(
+   filename: string
+): Promise<TestPlan> {
+   const data = await readContentFromFile(filename);
+
+   if (!data) {
+      new Error(`Cannot read test plan: ${filename}`);
+   }
+
+   const version = loadVersionToTest(data);
+
+   if (!version) {
+      new Error(`Cannot read Version to Test in ${filename}`);
+   }
+
+   const testPlanScenarios = await loadTestCases(data);
+
+   return {
+      versionToTest: version,
+      testCases: testPlanScenarios,
+      name: filename,
+   };
+}
+
+export function loadVersionToTest(data: string) {
+   const options = {
+      header: true,
+      columns: ["versionToTest"],
+      from_line: 1,
+      to_line: 2,
+      skip_empty_lines: true,
+      record_delimiter: RECORD_DELIMITER,
+   };
+   const versionFields = parse(data, options);
+
+   return !!versionFields[1] && versionFields[1].versionToTest.trim();
+}
+
+export async function loadTestCases(data: string): Promise<TestCase[]> {
+   const options = {
+      cast: (value: any, _context: any) => {
+         if (value === "no") {
```

Comment:
could this be simplified to `cast(value: any) => value !== "no"`?  I worry that the `return value` could lead to unexpected behavior because it would not be a true boolean.

Reply:
This cast is specific to the "yes" and "no" strings.  The goal is to cast only those, but returned all the others values unchanged.  It's some bit of black magic to turn a string back into an object.